models/engine/db_storage.py

```diff
@@ -52,19 +55,32 @@ def all(self, cls=None):
                     objects[obj.__class__.__name__ + '.' + obj.id] = obj
         return objects
 
+    def count(self, cls=None):
+        """Returns the number of objects in storage with given class name"""
+        return len(self.all(cls))
+
+    def get(self, cls, id):
+        """Returns the object based on the class name and its ID"""
+        key = cls + "." + id
+        return self.all().get(key, None)
```

Comment:
you could also use `self.__session.query(cls).filter(cls.id == id)`. That way you're not pulling in all your objects just to look for a single one. Could take a while to execute when the database starts getting filled.